assets/js/components/wp-dashboard/wp-dashboard-app.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WPDashboardApp component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Link from '../Link';
+import LegacyWPDashboardModules from './legacy-wp-dashboard-modules';
+import WPDashboardImpressions from './wp-dashboard-impressions';
+import WPDashboardClicks from './wp-dashboard-clicks';
+import WPDashboardUniqueVisitors from './wp-dashboard-unique-visitors';
+import WPDashboardSessionDuration from './wp-dashboard-session-duration';
+import AnalyticsInactiveCTA from '../AnalyticsInactiveCTA';
+import { STORE_NAME as CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+const { useSelect } = Data;
+
+const WPDashboardApp = () => {
+	const dashboardURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' ) );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
+	const analyticsModuleConnected = useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'analytics' ) );
+	const wpDashboardEnabled = featureFlags.widgets.wpDashboard.enabled;
+
+	if ( ! dashboardURL ) {
+		return null;
+	}
+
+	return (
+		<div className="googlesitekit-wp-dashboard">
+			<div className="googlesitekit-wp-dashboard__cta">
+				<Link className="googlesitekit-wp-dashboard__cta-link" href={ dashboardURL }>
+					{ __( 'Visit your Site Kit Dashboard', 'google-site-kit' ) }
+				</Link>
+			</div>
+			<div className="googlesitekit-wp-dashboard-stats googlesitekit-wp-dashboard-stats--fourup">
+				{ wpDashboardEnabled && (
+					<Fragment>
+						<WPDashboardImpressions />
+						<WPDashboardClicks />
+						{ ( analyticsModuleActive && analyticsModuleConnected ) ? (
+							<Fragment>
+								<WPDashboardUniqueVisitors />
+								<WPDashboardSessionDuration />
+							</Fragment>
+						) : (
+							<AnalyticsInactiveCTA />
+						) }
```

Comment:
Let's avoid ternaries for big sections of component markup like this wherever we can. It's harder to read and I think makes it less easier to read about the "if case" for something rendering or not.

```suggestion
						{ analyticsModuleActive && analyticsModuleConnected && (
							<Fragment>
								<WPDashboardUniqueVisitors />
								<WPDashboardSessionDuration />
							</Fragment>
						) }
						{ ( ! analyticsModuleActive || ! analyticsModuleConnected ) && <AnalyticsInactiveCTA /> }
```

That means we tie each section we render to a specific conditional—if either condition needs to change it can do so independently. It's what we've tried to use in general throughout the app and I think it's worth doing here too.

(Sidenote, I realise `( ! analyticsModuleActive || ! analyticsModuleConnected )` could be more concisely expressed with `! ( analyticsModuleActive && analyticsModuleConnected )` but the former is a bit clearer and I don't think that will lead to any bugs. 🙂 )